src/pint/models/model_builder.py

```diff
@@ -817,3 +915,149 @@ def get_model_and_toas(
         mm.add_tzr_toa(tt)
 
     return mm, tt
+
+
+def guess_binary_model(parfile_dict):
+    """Based on the PINT parameter dictionary, guess the binary model
+
+    Parameters
+    ----------
+    parfile_dict
+        The parameter dictionary as read-in by parse_parfile
+
+    Returns
+    -------
+    list:
+        A priority-ordered list of possible binary models. The first one is the
+        best-guess
+
+    """
+
+    def add_sini(parameters):
+        """If 'KIN' is a model parameter, Tempo2 doesn't really use SINI"""
+        if "KIN" in parameters:
+            return list(parameters) + ["SINI"]
+        else:
+            return list(parameters)
+
+    all_components = AllComponents()
+    binary_models = all_components.category_component_map["pulsar_system"]
+
+    # Find all binary parameters
+    # TODO: Check whetehr binary_model[:6]=="Binary" ??
+    binary_parameters_map = {
+        binary_model[6:]: add_sini(
```

Comment:
What does the index 6 signify? Is it OK to hardcode this value?

Reply:
Oh, right...  thanks for pointing this out. I have been meaning to ask you. In the parfile we name the binary models just with DD, BT, etc. But in PINT they are called BinaryBT, BinaryDD, etc. The 6 indicates the number of characters in "Binary". What do you think is the right way to do this? Do we always use the same naming convention here? In principle that is not necessary, so this is probably hacky and not the right way to do this.

Reply:
Maybe replace 6 with `len("Binary")`? This will clarify your intent at least.

Reply:
Actually, the binary classes have a `binary_model_name` attribute. This will have the desired strings like ELL1, DD etc.

Reply:
Thanks!